### ISSUE TITLE
No longer shows executions

### ISSUE THREAD

**Issue by wildsonc** [n8n-io/n8n#7373]
**Describe the bug**

From version `1.8.2` onwards, the execution log only shows the running status or manual executions, no other status, the API follows the same behavior.

As of version `1.9.0`, logs are no longer pruned, only soft delete, but in my instance I did not see them being deleted, reaching millions of lines ignoring the `EXECUTIONS_DATA_PRUNE_MAX_COUNT` flag.

My theory is that this new soft delete routine is ignoring all configurations, at the instance and workflow level, and therefore the API is not showing the executions correctly as they were soft deleted.

![CleanShot 2023-10-06 at 14 04 41@2x](https://github.com/n8n-io/n8n/assets/72408418/e99ab33b-51b2-486c-a49a-76aae596e44a)
These failed executions should have been saved, but they were "removed".

Maybe these changes didn't happen exactly in the ones I mentioned, but they were where I noticed these differences.

With version `1.10.0` I will monitor if the executions will be deleted.

**Environment (please complete the following information):**

- OS: `Docker`
- n8n Version: `1.10.0`
- Database system: `Postgres`
- Operation mode: `queue`

**Comment by Joffcom** [comment]
Hey @wildsonc,

I have just taken a look at my local install of n8n and it is correctly logging and displaying executions for me, I am not too sure on the soft delete side of things though I will find out more about that today.

**Comment by Joffcom** [comment]
@wildsonc is it possible that the issue you are seeing was fixed in `1.9.1` with https://github.com/n8n-io/n8n/pull/7333?

**Comment by wildsonc** [comment]
Hi @Joffcom . I'm already using version `1.10.0`, in this version pruning is working correctly, but it's still deleting failed executions that it shouldn't, if I change the switch to save successful executions it works as expected.

I added an error flow and in it I can see that it was executed, but in the UI it does not appear because they were soft-deleted.

![CleanShot 2023-10-09 at 07 15 43](https://github.com/n8n-io/n8n/assets/72408418/c8cb4b67-8c07-47d1-bf0e-2f61af10d401)
![CleanShot 2023-10-09 at 07 16 38](https://github.com/n8n-io/n8n/assets/72408418/6925a6f8-0ab8-4597-982b-3925584be9bd)

In the database I see that they were deleted as soon as they were finished.
<img width="1048" alt="image" src="https://github.com/n8n-io/n8n/assets/72408418/5cc76012-bc56-4b7f-a037-c00cedc2f87a">

If I change it to null in the deletedAt column, the executions appear again
![image](https://github.com/n8n-io/n8n/assets/72408418/0a4ab067-0430-4735-8470-ae9a6a8bcedf)

**Comment by Joffcom** [comment]
Hey @wildsonc,

When you say the below... 

> but it's still deleting failed executions that it shouldn't, if I change the switch to save successful executions it works as expected.

What do you actually mean? For my test I set my workflwos to save successful exections but it sounds like that isn't what you are doing so is the issue that failed executions are not showing but everything else is?

**Comment by wildsonc** [comment]
Yep, just the save failed executions setting doesn't work.

**Comment by Joffcom** [comment]
Perfect, Let me see if I can reproduce that.

**Comment by wildsonc** [comment]
I did some more tests with the workflow settings:

Save failed `all` | Save successful `all`:

- **Failed execution**: _Saved_ ✅
- **Successful execution**: _Saved_ ✅

Save failed `none` | Save successful `all`:

- **Failed execution**: _Saved_ ❌
- **Successful execution**: _Saved_ ✅

Save failed `all` | Save successful `none`:

- **Failed execution**: _Not Saved_ ❌
- **Successful execution**: _Not Saved_ ❌

Save failed `none` | Save successful `none`:

- **Failed execution**: _Not Saved_ ✅
- **Successful execution**: _Not Saved_ ✅

In short, the save on failure setting is being ignored, and the save on success option saves everything. When I say they were not saved I mean that they were soft-deleted in the database.

I found this [change](https://github.com/n8n-io/n8n/pull/7138/files#diff-f1214419cabb6abb8ec771d5f35be7d8059f52cae9dd610562119fa452490e3fL599) in the code, and it seems to be the root of the problem, where, from what I understand and I could be wrong, before it checked for an error in the node and now it expects a promise error, I didn't look deeper if when an error in a node throw in an error in the promise, but assuming not, this would explain these results where `workflowDidSucceed` is always `true`.

```diff
- const workflowDidSucceed = !runData.data.resultData.error;
+ const workflowDidSucceed = !racingPromisesResult.error;
if (
  (workflowDidSucceed && saveDataSuccessExecution === "none") ||
  (!workflowDidSucceed && saveDataErrorExecution === "none")
) {
  await Container.get(ExecutionRepository).softDelete(executionId);
}
```

**Comment by wildsonc** [comment]
I did some more tests with the workflow settings:

Save failed `all` | Save successful `all`:

- **Failed execution**: _Saved_ ✅
- **Successful execution**: _Saved_ ✅

Save failed `none` | Save successful `all`:

- **Failed execution**: _Saved_ ❌
- **Successful execution**: _Saved_ ✅

Save failed `all` | Save successful `none`:

- **Failed execution**: _Not Saved_ ❌
- **Successful execution**: _Not Saved_ ✅

Save failed `none` | Save successful `none`:

- **Failed execution**: _Not Saved_ ✅
- **Successful execution**: _Not Saved_ ✅

In short, the save on failure setting is being ignored, and the save on success option saves everything. When I say they were not saved I mean that they were soft-deleted in the database.

I found this [change](https://github.com/n8n-io/n8n/pull/7138/files#diff-f1214419cabb6abb8ec771d5f35be7d8059f52cae9dd610562119fa452490e3fL599) in the code, and it seems to be the root of the problem, where, from what I understand and I could be wrong, before it checked for an error in the node and now it expects a promise error, I didn't look deeper if when an error in a node throw in an error in the promise, but assuming not, this would explain these results where `workflowDidSucceed` is always `true`.

```diff
- const workflowDidSucceed = !runData.data.resultData.error;
+ const workflowDidSucceed = !racingPromisesResult.error;
if (
  (workflowDidSucceed && saveDataSuccessExecution === "none") ||
  (!workflowDidSucceed && saveDataErrorExecution === "none")
) {
  await Container.get(ExecutionRepository).softDelete(executionId);
}
```

**Comment by Joffcom** [comment]
Hey @wildsonc,

That table is perfect I can test that this morning quickly and get the dev ticket written up.

**Comment by Joffcom** [comment]
Hey @wildsonc,

I have created a workflow that has 2 nodes > Schedule (every 10 seconds) and Stop and Error.

Using this workflow I have then followed the same test sets with very different results.

**Save failed all | Save successful all:**
    Failed execution: Saved ✅
    Successful execution: Saved ✅

**Save failed none | Save successful all:**
    Failed execution: Not Saved ✅
    Successful execution: Saved ✅

**Save failed all | Save successful none:**
    Failed execution: Saved ✅
    Successful execution: Not Saved ✅

**Save failed none | Save successful none:**
    Failed execution: Not Saved ✅
    Successful execution: Not Saved ✅
 
What I did then realise was I not using queue mode and when I swapped to an instance using queue mode I was able to reproduce it exactly how you documented it, I should have paid more attention to the initial ticket... The internal dev ticket for this is `N8N-7006` and we should hopefully have it resolved soon.

**Comment by wildsonc** [comment]
Thanks @Joffcom.

**Comment by Joffcom** [comment]
Hey @wildsonc,

It looks like we fixed this in `1.14.2` in this pr: https://github.com/n8n-io/n8n/pull/7481 For now I am going to get this one closed and if you are still seeing an issue after updating let me know and we can open it again.

**Comment by wildsonc** [comment]
@Joffcom, I tested it now on version `1.15.1` and it still has the same problem.
